[Bug]: Panic when creating a new referral set without a team

### Problem encountered

Panic when creating a new referral set without a team

Referral set creation transaction:

```
[2023-09-12T11:08:53.262Z] 2023-09-12 11:08:52.953:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[   signtx_v2]:[ 388] Calling jsonrpc client.send_transaction for wallet: 5234813D87, pubkey: c4eeb702e8bf7d32212f881748909d0e71432a43430577c86ded75abc1a7054b: req: {'create_referral_set': {'is_team': False}, 'propagate': True, 'pubKey': 'c4eeb702e8bf7d32212f881748909d0e71432a43430577c86ded75abc1a7054b'}, epoch: 10, block: 103
[2023-09-12T11:08:53.262Z] 2023-09-12 11:08:52.953:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[     _header]:[  57] using token: AjJDVhbHzZQC7JePSGbd4ZZuu1Ko8IFwHvMyq8xP4QfnLA5MkHizzkPfyDNH22p6
[2023-09-12T11:08:53.262Z] 2023-09-12 11:08:52.956:[INFO] [MainProcess] [MainThread] [   clientv2.py]:[   signtx_v2]:[ 403] response: {'jsonrpc': '2.0', 'result': {'receivedAt': '2023-09-12T11:08:52.954618653Z', 'sentAt': '2023-09-12T11:08:52.95530603Z', 'transactionHash': '325864E84D0F9CA6731B48D2EEAB6E91BD11D13939CC1C1AC71451BD6D4AAA68', 'transaction': {'input_data': 'CKP5qZzOjJLZIRBn0j8A', 'signature': {'value': '1fb34adfa8580f14793f565a1028a3ea0c0b0b2cbcd7a97af462d44d76d3d2447532b9d4f6edb5447834467cc8387e3fe4479cd856959c0e0ce20f9f5266e70f', 'algo': 'vega/ed25519', 'version': 1}, 'From': {'PubKey': 'c4eeb702e8bf7d32212f881748909d0e71432a43430577c86ded75abc1a7054b'}, 'version': 3, 'pow': {'tid': '976797cbaf1c55c287afeceb72058ad71e86a7d055abf430efa0f9562b94686f', 'nonce': 7}}}, 'id': 'request'}
[2023-09-12T11:08:53.262Z] 2023-09-12 11:08:52.956:[INFO] [MainProcess] [MainThread] [   teamsMod.py]:[ create_team]:[  53] Transaction to create a team successfully sent: tx_hash='325864E84D0F9CA6731B48D2EEAB6E91BD11D13939CC1C1AC71451BD6D4AAA68'

```


```
Trace from file: /jenkins/workspace/common/system-tests-wrapper/networkdata/testnet/logs/testnet-nodeset-full-2-full/data-node-2.stderr-2023-09-12T11:07:52Z.log

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x293f45a]

goroutine 648 [running]:
code.vegaprotocol.io/vega/core/events.RefereeJoinedReferralSetEventFromStream({0x5a1d5d0?, 0xc00007e0f0?}, 0xc000048000)
	/jenkins/workspace/common/system-tests-wrapper/vega/core/events/referral_set.go:119 +0x9a
code.vegaprotocol.io/vega/datanode/broker.toEvent({0x5a1d5d0?, 0xc00007e0f0?}, 0x138?)
	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/convert.go:159 +0x6e5
code.vegaprotocol.io/vega/datanode/broker.deserializeEvent({0xc000e40788, 0x138, 0x138})
	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/deserialize.go:81 +0xcf
code.vegaprotocol.io/vega/datanode/broker.(*Deserializer).Receive.func1()
	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/deserialize.go:42 +0x24d
created by code.vegaprotocol.io/vega/datanode/broker.(*Deserializer).Receive
	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/deserialize.go:32 +0x118
```

### Observed behaviour

Panic when creating a new referral set without a team

### Expected behaviour

No panic and the referral set should be created successfully.

### Steps to reproduce

```bash
Run the below system test

System Test Name - test_create_referral_set_without_team
System Test Branch - feat/teams-tests
```


### Software version

vega - develop

### Failing test

test_create_referral_set_without_team

### Jenkins run

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/7704/

### Configuration used

_No response_

### Relevant log output

```shell
https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/7704/consoleText


[2023-09-12T11:09:19.587Z]         else:
[2023-09-12T11:09:19.587Z] >           raise _InactiveRpcError(state)  # pytype: disable=not-instantiable
[2023-09-12T11:09:19.587Z] E           grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
[2023-09-12T11:09:19.587Z] E           	status = StatusCode.UNAVAILABLE
[2023-09-12T11:09:19.587Z] E           	details = "failed to connect to all addresses; last error: UNKNOWN: ipv4:127.0.0.1:3027: Failed to connect to remote host: Connection refused"
[2023-09-12T11:09:19.587Z] E           	debug_error_string = "UNKNOWN:failed to connect to all addresses; last error: UNKNOWN: ipv4:127.0.0.1:3027: Failed to connect to remote host: Connection refused {grpc_status:14, created_time:"2023-09-12T11:09:08.240850581+00:00"}"
[2023-09-12T11:09:19.587Z] E           >
[2023-09-12T11:09:19.587Z] 
[2023-09-12T11:09:19.587Z] .venv/lib/python3.11/site-packages/grpc/_channel.py:910: _InactiveRpcError
[2023-09-12T11:09:19.587Z] -------------------- ERROR FOUND WITH NETWORK. ERROR 1 of 1 --------------------
[2023-09-12T11:09:19.587Z] Trace from file: /jenkins/workspace/common/system-tests-wrapper/networkdata/testnet/logs/testnet-nodeset-full-2-full/data-node-2.stderr-2023-09-12T11:07:52Z.log
[2023-09-12T11:09:19.587Z] 
[2023-09-12T11:09:19.587Z] panic: runtime error: invalid memory address or nil pointer dereference
[2023-09-12T11:09:19.587Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x293f45a]
[2023-09-12T11:09:19.587Z] 
[2023-09-12T11:09:19.587Z] goroutine 648 [running]:
[2023-09-12T11:09:19.587Z] code.vegaprotocol.io/vega/core/events.RefereeJoinedReferralSetEventFromStream({0x5a1d5d0?, 0xc00007e0f0?}, 0xc000048000)
[2023-09-12T11:09:19.587Z] 	/jenkins/workspace/common/system-tests-wrapper/vega/core/events/referral_set.go:119 +0x9a
[2023-09-12T11:09:19.587Z] code.vegaprotocol.io/vega/datanode/broker.toEvent({0x5a1d5d0?, 0xc00007e0f0?}, 0x138?)
[2023-09-12T11:09:19.587Z] 	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/convert.go:159 +0x6e5
[2023-09-12T11:09:19.587Z] code.vegaprotocol.io/vega/datanode/broker.deserializeEvent({0xc000e40788, 0x138, 0x138})
[2023-09-12T11:09:19.587Z] 	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/deserialize.go:81 +0xcf
[2023-09-12T11:09:19.587Z] code.vegaprotocol.io/vega/datanode/broker.(*Deserializer).Receive.func1()
[2023-09-12T11:09:19.587Z] 	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/deserialize.go:42 +0x24d
[2023-09-12T11:09:19.587Z] created by code.vegaprotocol.io/vega/datanode/broker.(*Deserializer).Receive
[2023-09-12T11:09:19.587Z] 	/jenkins/workspace/common/system-tests-wrapper/vega/datanode/broker/deserialize.go:32 +0x118
```

--- a/core/events/bus.go
+++ b/core/events/bus.go
@@ -315,7 +315,7 @@ var (
 		ReferralProgramStartedEvent:       eventspb.BusEventType_BUS_EVENT_TYPE_REFERRAL_PROGRAM_STARTED,
 		ReferralProgramEndedEvent:         eventspb.BusEventType_BUS_EVENT_TYPE_REFERRAL_PROGRAM_ENDED,
 		ReferralProgramUpdatedEvent:       eventspb.BusEventType_BUS_EVENT_TYPE_REFERRAL_PROGRAM_UPDATED,
-		ReferralSetCreatedEvent:           eventspb.BusEventType_BUS_EVENT_TYPE_REFEREE_JOINED_REFERRAL_SET,
+		ReferralSetCreatedEvent:           eventspb.BusEventType_BUS_EVENT_TYPE_REFERRAL_SET_CREATED,
 		RefereeJoinedReferralSetEvent:     eventspb.BusEventType_BUS_EVENT_TYPE_REFEREE_JOINED_REFERRAL_SET,
 		PartyActivityStreakEvent:          eventspb.BusEventType_BUS_EVENT_TYPE_PARTY_ACTIVITY_STREAK,
 		VolumeDiscountProgramStartedEvent: eventspb.BusEventType_BUS_EVENT_TYPE_VOLUME_DISCOUNT_PROGRAM_STARTED,

